### ISSUE TITLE
Potential further Lucene changes after #1587

### ISSUE THREAD

**Issue by jarohen** [xtdb/xtdb#1599]
_Originally posted by @refset in https://github.com/juxt/crux/issues/1587#issuecomment-893677499_

I will open a separate PR for other nice-to-have changes later:

- [x] ~~#1654 respect main query timeout (at a minimum could pass outer timeout into Lucene)~~
- [ ] re-using Fields and Document, except unlike the commit in this PR, only add/remove the one changing field (just use existing structure mutably)
- [ ] add index-version to indexer protocol, takes a string for descriptive versioning ("eavindexer-1"), bump lucene index version(s) to 2 - this can happen just before the next version bump and can be a separate protocol for incremental change
- [ ] configurable `search*` batch size - `{:keys [batch-size]}` opts map with two-arity versions
- [ ] microbench `DocumentId` change, e.g. current Record+Nippy, vs `id-str (str (count e-str) "." e-str (count k-str) "." k-str v)` (which is a more correct version that the commit in this PR)
- [ ] use byte array instead of str for `->hash ids`, probably only in place of e-str and id-str (looking at this PR)
